### ISSUE TITLE
Popup theme, wrong tabid and no antiForgeryToken

### ISSUE THREAD

**Issue by valadas** [dnnsoftware/Dnn.Platform#2548]
I am developing an angular app, in MVC, using web api, and this works fine in a NON popup theme, such as Admin and Home (Xcillion). In a popup theme the tabid is set to -1 and there is no element RequestVerificationToken and thus using sf to get the antiForgeryToken OR window.document.getElementsByName('__RequestVerificationToken')[0].value is not possible; this occurs only for non administrators. When logged in as super user this issue is not present.

In summary:

MVC, DNN popup theme and Non administrator: the above mentioned problem occurs. Using dnn9 with the default theme (Xcillion).
In other scenarios the tabid is set correctly and the antiForgeryToken is present.
Originally created by @kurtwilbies at dnnsoftware/Dnn.AdminExperience.Library#9

Moved here from https://github.com/dnnsoftware/Dnn.AdminExperience/issues/4 since this does not relate to the persona bar.

**Comment by bdukes** [comment]
@kurtwilbies, I know this has been a while since you originally logged the issue, so I don't know if you have an easy way to recreate it still, or if you've found a fix or workaround in the meantime.

I believe the issue is that you need to call [`ServicesFramework.Instance.RequestAjaxAntiForgerySupport`](https://www.dnnsoftware.com/dnn-api/?topic=html/b2de35b4-fd0e-ac51-f577-3e16a3fdcf00.htm) from your code to ensure that the tab ID and request verification token are available.  Since the popup theme is so minimal there's nothing else on the page that is causing that request, unless you're an admin.

If you have a quick way to validate that, let us know that it worked, but I understand that we're very late in getting back with you, so you might not be able to.

Hope it helps!

**Comment by kurtwilbies** [comment]
Hi, the fix is quite easy. You'll have to call servicesframework is your cshtml file. Not sure which one.

**Comment by valadas** [comment]
@kurtwilbies not sure I understand your answer, this issue was moved here by me but originally created by you in the AdminExperience repository. So do I get that you found your answer and we can close this issue or are you asking where this should be put?

**Comment by kurtwilbies** [comment]
Close the issue.
